2023-03-26 08:59:33,404 - INFO - [MainThread] - Initialized router P8 at port 5008 PID 16314
2023-03-26 08:59:33,408 - INFO - [MainThread] - Adding a connection to LupusreginaBeta
2023-03-26 08:59:33,408 - INFO - [Connect-LupusreginaBeta] - Calling connect_to_rendezvous_point
2023-03-26 08:59:33,408 - INFO - [Connect-LupusreginaBeta] - Connect to rendezvous point LupusreginaBeta
2023-03-26 08:59:33,408 - INFO - [MainThread] - Adding listener for Vulnerar
2023-03-26 08:59:33,411 - INFO - [Establish-Vulnerar] - Establishing RP to Vulnerar
2023-03-26 08:59:33,415 - INFO - [Connect-LupusreginaBeta] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 08:59:33,422 - INFO - [MainThread] - Starting testing thread
2023-03-26 08:59:33,430 - INFO - [Establish-Vulnerar] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 08:59:33,437 - INFO - [TESTER] - Started tester
2023-03-26 08:59:33,451 - INFO - [MainThread] - Starting connection thread
2023-03-26 08:59:33,466 - INFO - [MainThread] - Starting log thread
2023-03-26 08:59:33,880 - INFO - [Connect-LupusreginaBeta] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 08:59:33,909 - INFO - [Establish-Vulnerar] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 08:59:33,910 - INFO - [Establish-Vulnerar] - Connecting to guard node 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13)... (None)
2023-03-26 08:59:33,991 - INFO - [Connect-LupusreginaBeta] - Connected to rendezvous point LupusreginaBeta
2023-03-26 08:59:33,991 - INFO - [Connect-LupusreginaBeta] - Waiting for peer to open stream
2023-03-26 08:59:34,373 - INFO - [Connect-LupusreginaBeta] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-26 08:59:34,374 - INFO - [Connect-LupusreginaBeta] - Stream opened successfully
2023-03-26 08:59:34,374 - INFO - [Connect-LupusreginaBeta] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P8\x94ua.'
2023-03-26 08:59:34,658 - INFO - [Establish-Vulnerar] - Creating new circuit #80000001 with 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13) router...
2023-03-26 08:59:34,778 - INFO - [RECEIVER] - Added neighbour P6
2023-03-26 08:59:34,848 - INFO - [Establish-Vulnerar] - Building -1 hops circuit...
2023-03-26 08:59:34,848 - INFO - [Establish-Vulnerar] - Extending the circuit #80000001 with 157.90.179.103:8001 (Vulnerar; Tor 0.4.7.13)...
2023-03-26 08:59:34,849 - INFO - [Establish-Vulnerar] - Getting descriptor for 157.90.179.103:8001 (Vulnerar; Tor 0.4.7.13)...
2023-03-26 08:59:34,853 - INFO - [Establish-Vulnerar] - Connecting to guard node 86.160.180.137:9001 (Bates1; Tor 0.4.7.13)... (Internal dir client)
2023-03-26 08:59:35,057 - INFO - [Establish-Vulnerar] - Creating new circuit #80000002 with 86.160.180.137:9001 (Bates1; Tor 0.4.7.13) router...
2023-03-26 08:59:35,104 - INFO - [Establish-Vulnerar] - Building 0 hops circuit...
2023-03-26 08:59:35,104 - INFO - [Establish-Vulnerar] - Stream #1: creating attached to #80000002 circuit...
2023-03-26 08:59:35,104 - INFO - [Establish-Vulnerar] - Stream #1: connecting to hsdir
2023-03-26 08:59:35,148 - INFO - [RecvLoop_86.160.] - Stream #1: connected (remote ip '')
2023-03-26 08:59:35,399 - INFO - [RecvLoop_86.160.] - Stream #1: remote disconnected (reason = DONE)
2023-03-26 08:59:35,399 - INFO - [Establish-Vulnerar] - Stream #1: closing (state = Disconnected)...
2023-03-26 08:59:35,399 - INFO - [Establish-Vulnerar] - Got descriptor
2023-03-26 08:59:35,755 - INFO - [Establish-Vulnerar] - <class 'torpy.circuit.TorCircuit'>
2023-03-26 08:59:36,110 - INFO - [Establish-Vulnerar] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-26 08:59:36,111 - INFO - [Establish-Vulnerar] - Waiting for connections at relay Vulnerar for cookie b'eb688e4f52df90278061' ...
2023-03-26 08:59:37,887 - INFO - [Establish-Vulnerar] - Got REND2 message from Vulnerar
2023-03-26 08:59:37,887 - INFO - [Establish-Vulnerar] - Entered circuit context with id 0x80000001
2023-03-26 08:59:37,887 - INFO - [Establish-Vulnerar] - Stream #2: creating attached to #80000001 circuit...
2023-03-26 08:59:37,887 - INFO - [Establish-Vulnerar] - Stream #2: connecting to ('127.0.0.1', 5000)
2023-03-26 08:59:38,307 - INFO - [RecvLoop_97.113.] - Stream #2: connected (remote ip '127.0.0.1')
2023-03-26 08:59:38,506 - INFO - [Establish-Vulnerar] - Sending HELLO to peer
2023-03-26 08:59:38,507 - INFO - [Establish-Vulnerar] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P8\x94ua.'
2023-03-26 08:59:38,920 - INFO - [RECEIVER] - Added neighbour P7
2023-03-26 09:00:14,028 - WARNING - [Connect-LupusreginaBeta] - Got SENDME
2023-03-26 09:00:33,426 - WARNING - [RECEIVER] - Routing loop from P8 to P4: P8-P6-P8
2023-03-26 09:00:33,427 - WARNING - [RECEIVER] - Routing loop from P8 to P5: P8-P6-P8
2023-03-26 09:00:33,427 - WARNING - [RECEIVER] - Routing loop from P8 to P2: P8-P6-P8
2023-03-26 09:00:33,512 - INFO - [CONNECTION] - Neighbour connections (2) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:00:33,513 - INFO - [CONNECTION] - Candidate peers: ['P4', 'P5', 'P2', 'P1', 'P3']
2023-03-26 09:00:33,513 - INFO - [CONNECTION] - Highest cost peers: [('P2', 1582), ('P1', 1482), ('P3', 1382), ('P4', 884), ('P5', 200), ('P6', 100), ('P7', 100), ('P8', 0)]
2023-03-26 09:00:33,517 - INFO - [CONNECTION] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:00:33,751 - INFO - [CONNECTION] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:00:33,760 - INFO - [MainThread] - ['ESTABLISH', 'zwewwlUA1', b'fe1e1bbcc86d2ce997ca']
2023-03-26 09:00:33,760 - INFO - [MainThread] - Got ESTABLISH to relay zwewwlUA1:b'fe1e1bbcc86d2ce997ca' via NeelTorRelay2
2023-03-26 09:00:33,761 - INFO - [Establish-zwewwlUA1] - Establishing RP to zwewwlUA1
2023-03-26 09:00:33,765 - INFO - [Establish-zwewwlUA1] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:00:34,002 - INFO - [Establish-zwewwlUA1] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:00:34,003 - INFO - [Establish-zwewwlUA1] - Connecting to guard node 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13)... (None)
2023-03-26 09:00:34,764 - INFO - [Establish-zwewwlUA1] - Creating new circuit #80000003 with 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13) router...
2023-03-26 09:00:34,952 - INFO - [Establish-zwewwlUA1] - Building -1 hops circuit...
2023-03-26 09:00:34,952 - INFO - [Establish-zwewwlUA1] - Extending the circuit #80000003 with 217.12.221.75:9001 (zwewwlUA1; Tor 0.4.5.10)...
2023-03-26 09:00:34,952 - INFO - [Establish-zwewwlUA1] - Getting descriptor for 217.12.221.75:9001 (zwewwlUA1; Tor 0.4.5.10)...
2023-03-26 09:00:34,956 - INFO - [Establish-zwewwlUA1] - Connecting to guard node 217.79.181.76:443 (ForPrivacyNET; Tor 0.4.7.13)... (Internal dir client)
2023-03-26 09:00:35,156 - INFO - [Establish-zwewwlUA1] - Creating new circuit #80000004 with 217.79.181.76:443 (ForPrivacyNET; Tor 0.4.7.13) router...
2023-03-26 09:00:35,223 - INFO - [Establish-zwewwlUA1] - Building 0 hops circuit...
2023-03-26 09:00:35,223 - INFO - [Establish-zwewwlUA1] - Stream #3: creating attached to #80000004 circuit...
2023-03-26 09:00:35,223 - INFO - [Establish-zwewwlUA1] - Stream #3: connecting to hsdir
2023-03-26 09:00:35,270 - INFO - [RecvLoop_217.79.] - Stream #3: connected (remote ip '')
2023-03-26 09:00:35,472 - INFO - [RecvLoop_217.79.] - Stream #3: remote disconnected (reason = DONE)
2023-03-26 09:00:35,473 - INFO - [Establish-zwewwlUA1] - Stream #3: closing (state = Disconnected)...
2023-03-26 09:00:35,473 - INFO - [Establish-zwewwlUA1] - Got descriptor
2023-03-26 09:00:35,853 - INFO - [Establish-zwewwlUA1] - <class 'torpy.circuit.TorCircuit'>
2023-03-26 09:00:36,231 - INFO - [Establish-zwewwlUA1] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-26 09:00:36,232 - INFO - [Establish-zwewwlUA1] - Waiting for connections at relay zwewwlUA1 for cookie b'fe1e1bbcc86d2ce997ca' ...
2023-03-26 09:00:39,273 - INFO - [RecvLoop_86.160.] - On destroy: circuit #80000002
2023-03-26 09:00:39,273 - INFO - [RecvLoop_86.160.] - Destroy circuit #80000002
2023-03-26 09:00:43,777 - INFO - [CONNECTION] - Sending JOIN to P2 at relay zwewwlUA1
2023-03-26 09:00:44,790 - INFO - [RECEIVER] - Added relay gGDHjdcC6zAlM8k08lX from P3 to JOIN queue
2023-03-26 09:00:44,792 - INFO - [MainThread] - ['JOIN', 'gGDHjdcC6zAlM8k08lX', b'894269aca75c31563290']
2023-03-26 09:00:44,793 - INFO - [MainThread] - Got JOIN to relay gGDHjdcC6zAlM8k08lX, waiting 10 sec to connect
2023-03-26 09:00:44,793 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Calling connect_to_rendezvous_point
2023-03-26 09:00:44,793 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Connect to rendezvous point gGDHjdcC6zAlM8k08lX
2023-03-26 09:00:44,800 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:00:45,091 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:00:45,261 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Connected to rendezvous point gGDHjdcC6zAlM8k08lX
2023-03-26 09:00:45,261 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Waiting for peer to open stream
2023-03-26 09:00:45,336 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-26 09:00:45,337 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Stream opened successfully
2023-03-26 09:00:45,337 - INFO - [Connect-gGDHjdcC6zAlM8k08lX] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P8\x94ua.'
2023-03-26 09:00:45,798 - INFO - [RECEIVER] - Added neighbour P3
2023-03-26 09:00:46,396 - WARNING - [Connect-LupusreginaBeta] - Got SENDME
2023-03-26 09:00:46,577 - WARNING - [Connect-LupusreginaBeta] - Got SENDME
2023-03-26 09:01:20,836 - INFO - [RECEIVER] - updateGraphOnly: [['P1', 'P2', 768, '4mnesia'], ['P1', 'P3', 456, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 100, 'DTFNODE75'], ['P2', 'P4', 758, 'LupusreginaBeta'], ['P2', 'P5', 100, 'bauruine'], ['P2', 'P6', 100, 'AlexandruIoanCuza2'], ['P3', 'P4', 498, 'Vulnerar'], ['P3', 'P8', 100, 'gGDHjdcC6zAlM8k08lX'], ['P4', 'P5', 684, 'Cornball'], ['P5', 'P6', 897, 'taipe'], ['P5', 'P7', 1204, 'RMK'], ['P6', 'P8', 908, 'LupusreginaBeta'], ['P7', 'P8', 1444, 'Vulnerar']] , {'P8'}
2023-03-26 09:01:24,817 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:01:33,598 - INFO - [Thread-4] - Adding P6 to list of inactive
2023-03-26 09:01:33,598 - INFO - [Thread-4] - DEAD ROUTES DETECTED: P6
2023-03-26 09:01:33,598 - INFO - [Thread-4] - Removing P6 from neighbours
2023-03-26 09:01:33,598 - INFO - [Thread-4] - updateLSADB: removing {'P6'}
2023-03-26 09:01:33,598 - INFO - [Thread-4] - updateGraphOnly: [['P1', 'P2', 768, '4mnesia'], ['P1', 'P3', 456, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 100, 'DTFNODE75'], ['P2', 'P4', 758, 'LupusreginaBeta'], ['P2', 'P5', 100, 'bauruine'], ['P2', 'P6', 100, 'AlexandruIoanCuza2'], ['P3', 'P4', 498, 'Vulnerar'], ['P3', 'P8', 100, 'gGDHjdcC6zAlM8k08lX'], ['P4', 'P5', 684, 'Cornball'], ['P5', 'P6', 897, 'taipe'], ['P5', 'P7', 1204, 'RMK'], ['P7', 'P8', 1444, 'Vulnerar']] , {'P6'}
2023-03-26 09:01:37,398 - INFO - [RECEIVER] - Added relay TestMyTor721 from P1 to JOIN queue
2023-03-26 09:01:37,398 - INFO - [MainThread] - ['JOIN', 'TestMyTor721', b'fe1e1bbcc86d2ce997ca']
2023-03-26 09:01:37,398 - INFO - [MainThread] - Got JOIN to relay TestMyTor721, waiting 10 sec to connect
2023-03-26 09:01:37,399 - INFO - [Connect-TestMyTor721] - Calling connect_to_rendezvous_point
2023-03-26 09:01:37,399 - INFO - [Connect-TestMyTor721] - Connect to rendezvous point TestMyTor721
2023-03-26 09:01:37,403 - INFO - [Connect-TestMyTor721] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:01:37,592 - INFO - [Connect-TestMyTor721] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:01:38,677 - INFO - [Connect-TestMyTor721] - Connected to rendezvous point TestMyTor721
2023-03-26 09:01:38,678 - INFO - [Connect-TestMyTor721] - Waiting for peer to open stream
2023-03-26 09:01:39,108 - INFO - [Connect-TestMyTor721] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-26 09:01:39,108 - INFO - [Connect-TestMyTor721] - Stream opened successfully
2023-03-26 09:01:39,108 - INFO - [Connect-TestMyTor721] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P8\x94ua.'
2023-03-26 09:01:39,715 - INFO - [RECEIVER] - Added neighbour P1
2023-03-26 09:01:43,621 - INFO - [RecvLoop_217.79.] - On destroy: circuit #80000004
2023-03-26 09:01:43,621 - INFO - [RecvLoop_217.79.] - Destroy circuit #80000004
2023-03-26 09:01:43,840 - INFO - [CONNECTION] - Neighbour connections (3) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:01:43,840 - INFO - [CONNECTION] - Candidate peers: ['P2', 'P4', 'P5']
2023-03-26 09:01:43,840 - INFO - [CONNECTION] - Highest cost peers: [('P6', 1424), ('P5', 1340), ('P2', 1324), ('P7', 656), ('P4', 656), ('P1', 556), ('P3', 100), ('P8', 0)]
2023-03-26 09:01:43,844 - INFO - [CONNECTION] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:01:44,077 - INFO - [CONNECTION] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:01:44,084 - INFO - [MainThread] - ['ESTABLISH', 'testowov3', b'58cdec9f0feb2705127b']
2023-03-26 09:01:44,084 - INFO - [MainThread] - Got ESTABLISH to relay testowov3:b'58cdec9f0feb2705127b' via NeelTorRelay2
2023-03-26 09:01:44,084 - INFO - [Establish-testowov3] - Establishing RP to testowov3
2023-03-26 09:01:44,091 - INFO - [Establish-testowov3] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:01:44,284 - INFO - [Establish-testowov3] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:01:44,285 - INFO - [Establish-testowov3] - Connecting to guard node 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13)... (None)
2023-03-26 09:01:45,041 - INFO - [Establish-testowov3] - Creating new circuit #80000005 with 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13) router...
2023-03-26 09:01:45,235 - INFO - [Establish-testowov3] - Building -1 hops circuit...
2023-03-26 09:01:45,235 - INFO - [Establish-testowov3] - Extending the circuit #80000005 with 217.174.247.196:9998 (testowov3; Tor 0.4.7.13)...
2023-03-26 09:01:45,235 - INFO - [Establish-testowov3] - Getting descriptor for 217.174.247.196:9998 (testowov3; Tor 0.4.7.13)...
2023-03-26 09:01:45,241 - INFO - [Establish-testowov3] - Connecting to guard node 5.181.51.6:9001 (torrelay; Tor 0.4.7.10)... (Internal dir client)
2023-03-26 09:01:45,384 - INFO - [Establish-testowov3] - Creating new circuit #80000006 with 5.181.51.6:9001 (torrelay; Tor 0.4.7.10) router...
2023-03-26 09:01:45,420 - INFO - [Establish-testowov3] - Building 0 hops circuit...
2023-03-26 09:01:45,421 - INFO - [Establish-testowov3] - Stream #4: creating attached to #80000006 circuit...
2023-03-26 09:01:45,421 - INFO - [Establish-testowov3] - Stream #4: connecting to hsdir
2023-03-26 09:01:45,456 - INFO - [RecvLoop_5.181.5] - Stream #4: connected (remote ip '')
2023-03-26 09:01:45,700 - INFO - [RecvLoop_5.181.5] - Stream #4: remote disconnected (reason = DONE)
2023-03-26 09:01:45,700 - INFO - [Establish-testowov3] - Stream #4: closing (state = Disconnected)...
2023-03-26 09:01:45,700 - INFO - [Establish-testowov3] - Got descriptor
2023-03-26 09:01:46,041 - INFO - [Establish-testowov3] - <class 'torpy.circuit.TorCircuit'>
2023-03-26 09:01:46,378 - INFO - [Establish-testowov3] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-26 09:01:46,378 - INFO - [Establish-testowov3] - Waiting for connections at relay testowov3 for cookie b'58cdec9f0feb2705127b' ...
2023-03-26 09:01:54,086 - INFO - [CONNECTION] - Sending JOIN to P5 at relay testowov3
2023-03-26 09:01:56,172 - WARNING - [RECEIVER] - Routing loop from P8 to P7: P8-P3-P8
2023-03-26 09:01:56,173 - WARNING - [RECEIVER] - Routing loop from P8 to P5: P8-P3-P8
2023-03-26 09:02:01,530 - INFO - [RECEIVER] - Added relay DoubleX from P5 to JOIN queue
2023-03-26 09:02:01,531 - INFO - [MainThread] - ['JOIN', 'DoubleX', b'58cdec9f0feb2705127b']
2023-03-26 09:02:01,531 - INFO - [MainThread] - Got JOIN to relay DoubleX, waiting 10 sec to connect
2023-03-26 09:02:01,531 - INFO - [Connect-DoubleX] - Calling connect_to_rendezvous_point
2023-03-26 09:02:01,531 - INFO - [Connect-DoubleX] - Connect to rendezvous point DoubleX
2023-03-26 09:02:01,538 - INFO - [Connect-DoubleX] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:02:01,823 - INFO - [Connect-DoubleX] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:02:02,163 - ERROR - [Connect-LupusreginaBeta] - Got DESTROY
2023-03-26 09:02:04,354 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:02:04,816 - INFO - [Connect-DoubleX] - Connected to rendezvous point DoubleX
2023-03-26 09:02:04,816 - INFO - [Connect-DoubleX] - Waiting for peer to open stream
2023-03-26 09:02:05,868 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:02:06,194 - INFO - [Connect-DoubleX] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-26 09:02:06,194 - INFO - [Connect-DoubleX] - Stream opened successfully
2023-03-26 09:02:06,195 - INFO - [Connect-DoubleX] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P8\x94ua.'
2023-03-26 09:02:10,349 - INFO - [RECEIVER] - Added neighbour P5
2023-03-26 09:02:32,015 - WARNING - [Connect-TestMyTor721] - Got SENDME
2023-03-26 09:02:36,836 - WARNING - [RECEIVER] - Routing loop from P8 to P5: P8-P1-P8
2023-03-26 09:02:51,423 - INFO - [RecvLoop_5.181.5] - On destroy: circuit #80000006
2023-03-26 09:02:51,423 - INFO - [RecvLoop_5.181.5] - Destroy circuit #80000006
2023-03-26 09:02:54,136 - INFO - [CONNECTION] - Neighbour connections (4) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:02:54,136 - INFO - [CONNECTION] - Candidate peers: ['P2', 'P4']
2023-03-26 09:02:54,136 - INFO - [CONNECTION] - Highest cost peers: [('P6', 1424), ('P3', 300), ('P7', 200), ('P4', 200), ('P2', 200), ('P5', 100), ('P1', 100), ('P8', 0)]
2023-03-26 09:02:54,139 - INFO - [CONNECTION] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:02:54,350 - INFO - [CONNECTION] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:02:54,362 - INFO - [MainThread] - ['ESTABLISH', 'tirz', b'167145b6af4f691af9f5']
2023-03-26 09:02:54,362 - INFO - [MainThread] - Got ESTABLISH to relay tirz:b'167145b6af4f691af9f5' via NeelTorRelay2
2023-03-26 09:02:54,363 - INFO - [Establish-tirz] - Establishing RP to tirz
2023-03-26 09:02:54,369 - INFO - [Establish-tirz] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:02:54,597 - INFO - [Establish-tirz] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:02:54,600 - INFO - [Establish-tirz] - Connecting to guard node 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13)... (None)
2023-03-26 09:02:55,372 - INFO - [Establish-tirz] - Creating new circuit #80000007 with 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13) router...
2023-03-26 09:02:55,569 - INFO - [Establish-tirz] - Building -1 hops circuit...
2023-03-26 09:02:55,569 - INFO - [Establish-tirz] - Extending the circuit #80000007 with 51.158.230.172:443 (tirz; Tor 0.4.7.13)...
2023-03-26 09:02:55,569 - INFO - [Establish-tirz] - Getting descriptor for 51.158.230.172:443 (tirz; Tor 0.4.7.13)...
2023-03-26 09:02:55,575 - INFO - [Establish-tirz] - Connecting to guard node 82.65.28.207:443 (MayoiHachikuji; Tor 0.4.7.13)... (Internal dir client)
2023-03-26 09:02:55,847 - INFO - [Establish-tirz] - Creating new circuit #80000008 with 82.65.28.207:443 (MayoiHachikuji; Tor 0.4.7.13) router...
2023-03-26 09:02:55,921 - INFO - [Establish-tirz] - Building 0 hops circuit...
2023-03-26 09:02:55,921 - INFO - [Establish-tirz] - Stream #5: creating attached to #80000008 circuit...
2023-03-26 09:02:55,921 - INFO - [Establish-tirz] - Stream #5: connecting to hsdir
2023-03-26 09:02:55,978 - INFO - [RecvLoop_82.65.2] - Stream #5: connected (remote ip '')
2023-03-26 09:02:56,234 - INFO - [RecvLoop_82.65.2] - Stream #5: remote disconnected (reason = DONE)
2023-03-26 09:02:56,235 - INFO - [Establish-tirz] - Stream #5: closing (state = Disconnected)...
2023-03-26 09:02:56,235 - INFO - [Establish-tirz] - Got descriptor
2023-03-26 09:02:56,586 - INFO - [Establish-tirz] - <class 'torpy.circuit.TorCircuit'>
2023-03-26 09:02:56,928 - INFO - [Establish-tirz] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-26 09:02:56,928 - INFO - [Establish-tirz] - Waiting for connections at relay tirz for cookie b'167145b6af4f691af9f5' ...
2023-03-26 09:03:03,506 - WARNING - [Connect-DoubleX] - Got SENDME
2023-03-26 09:03:04,363 - INFO - [CONNECTION] - Sending JOIN to P4 at relay tirz
2023-03-26 09:03:04,374 - WARNING - [Connect-TestMyTor721] - Got SENDME
2023-03-26 09:03:04,628 - WARNING - [Connect-TestMyTor721] - Got SENDME
2023-03-26 09:03:12,717 - INFO - [Establish-tirz] - Got REND2 message from tirz
2023-03-26 09:03:12,718 - INFO - [Establish-tirz] - Entered circuit context with id 0x80000007
2023-03-26 09:03:12,718 - INFO - [Establish-tirz] - Stream #6: creating attached to #80000007 circuit...
2023-03-26 09:03:12,718 - INFO - [Establish-tirz] - Stream #6: connecting to ('127.0.0.1', 5000)
2023-03-26 09:03:13,101 - INFO - [RecvLoop_97.113.] - Stream #6: connected (remote ip '127.0.0.1')
2023-03-26 09:03:13,122 - INFO - [Establish-tirz] - Sending HELLO to peer
2023-03-26 09:03:13,122 - INFO - [Establish-tirz] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P8\x94ua.'
2023-03-26 09:03:13,971 - INFO - [RECEIVER] - Added neighbour P4
2023-03-26 09:03:18,719 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:03:31,381 - WARNING - [Connect-TestMyTor721] - Got SENDME
2023-03-26 09:03:36,923 - WARNING - [Connect-DoubleX] - Got SENDME
2023-03-26 09:03:36,923 - WARNING - [Connect-DoubleX] - Got SENDME
2023-03-26 09:03:56,904 - INFO - [RecvLoop_82.65.2] - On destroy: circuit #80000008
2023-03-26 09:03:56,904 - INFO - [RecvLoop_82.65.2] - Destroy circuit #80000008
2023-03-26 09:04:04,425 - INFO - [CONNECTION] - Neighbour connections (5) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:04:04,429 - INFO - [CONNECTION] - Candidate peers: ['P2']
2023-03-26 09:04:04,429 - INFO - [CONNECTION] - Highest cost peers: [('P6', 1424), ('P7', 200), ('P2', 200), ('P3', 200), ('P5', 100), ('P4', 100), ('P1', 100), ('P8', 0)]
2023-03-26 09:04:04,437 - INFO - [CONNECTION] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:04:04,929 - INFO - [CONNECTION] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:04:04,936 - INFO - [MainThread] - ['ESTABLISH', 'IceMateria', b'714e1f9b1809d0daf19a']
2023-03-26 09:04:04,936 - INFO - [MainThread] - Got ESTABLISH to relay IceMateria:b'714e1f9b1809d0daf19a' via NeelTorRelay2
2023-03-26 09:04:04,937 - INFO - [Establish-IceMateria] - Establishing RP to IceMateria
2023-03-26 09:04:04,953 - INFO - [Establish-IceMateria] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-26 09:04:05,242 - INFO - [Establish-IceMateria] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-26 09:04:05,244 - INFO - [Establish-IceMateria] - Connecting to guard node 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13)... (None)
2023-03-26 09:04:06,004 - INFO - [Establish-IceMateria] - Creating new circuit #80000009 with 97.113.253.84:110 (NeelTorRelay2; Tor 0.4.7.13) router...
2023-03-26 09:04:06,195 - INFO - [Establish-IceMateria] - Building -1 hops circuit...
2023-03-26 09:04:06,195 - INFO - [Establish-IceMateria] - Extending the circuit #80000009 with 194.76.227.150:9001 (IceMateria; Tor 0.4.7.13)...
2023-03-26 09:04:06,196 - INFO - [Establish-IceMateria] - Getting descriptor for 194.76.227.150:9001 (IceMateria; Tor 0.4.7.13)...
2023-03-26 09:04:06,201 - INFO - [Establish-IceMateria] - Connecting to guard node 192.42.115.101:9003 (cry; Tor 0.4.7.13)... (Internal dir client)
2023-03-26 09:04:06,372 - INFO - [Establish-IceMateria] - Creating new circuit #8000000a with 192.42.115.101:9003 (cry; Tor 0.4.7.13) router...
2023-03-26 09:04:06,432 - INFO - [Establish-IceMateria] - Building 0 hops circuit...
2023-03-26 09:04:06,432 - INFO - [Establish-IceMateria] - Stream #7: creating attached to #8000000a circuit...
2023-03-26 09:04:06,432 - INFO - [Establish-IceMateria] - Stream #7: connecting to hsdir
2023-03-26 09:04:06,467 - INFO - [RecvLoop_192.42.] - Stream #7: connected (remote ip '')
2023-03-26 09:04:06,685 - INFO - [RecvLoop_192.42.] - Stream #7: remote disconnected (reason = DONE)
2023-03-26 09:04:06,686 - INFO - [Establish-IceMateria] - Stream #7: closing (state = Disconnected)...
2023-03-26 09:04:06,686 - INFO - [Establish-IceMateria] - Got descriptor
2023-03-26 09:04:07,072 - INFO - [Establish-IceMateria] - <class 'torpy.circuit.TorCircuit'>
2023-03-26 09:04:07,211 - INFO - [RECEIVER] - updateLSADB: removing {'P3'}
2023-03-26 09:04:07,212 - INFO - [RECEIVER] - updateGraphOnly: [['P1', 'P2', 1175, '4mnesia'], ['P1', 'P3', 1418, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 2290, 'DTFNODE75'], ['P1', 'P8', 100, 'TestMyTor721'], ['P2', 'P4', 885, 'LupusreginaBeta'], ['P2', 'P5', 100, 'bauruine'], ['P3', 'P4', 1969, 'Vulnerar'], ['P3', 'P8', 745, 'gGDHjdcC6zAlM8k08lX'], ['P3', 'P7', 1984, 'honestmistake'], ['P3', 'P5', 100, 'orangeneis'], ['P4', 'P5', 783, 'Cornball'], ['P4', 'P7', 100, 'schmortor'], ['P4', 'P8', 100, 'tirz'], ['P5', 'P7', 1904, 'RMK'], ['P5', 'P8', 100, 'DoubleX'], ['P7', 'P8', 2680, 'Vulnerar']] , {'P3'}
2023-03-26 09:04:07,281 - WARNING - [Connect-DoubleX] - Got SENDME
2023-03-26 09:04:07,501 - INFO - [Establish-IceMateria] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-26 09:04:07,501 - INFO - [Establish-IceMateria] - Waiting for connections at relay IceMateria for cookie b'714e1f9b1809d0daf19a' ...
2023-03-26 09:04:13,946 - WARNING - [RECEIVER] - Routing loop from P8 to P3: P8-P5-P8
2023-03-26 09:04:14,955 - INFO - [CONNECTION] - Sending JOIN to P2 at relay IceMateria
2023-03-26 09:04:21,438 - WARNING - [Connect-TestMyTor721] - Got SENDME
2023-03-26 09:04:23,348 - WARNING - [Connect-TestMyTor721] - Got SENDME
2023-03-26 09:04:27,593 - INFO - [RECEIVER] - updateLSADB: removing {'P4'}
2023-03-26 09:04:27,594 - INFO - [RECEIVER] - updateGraphOnly: [['P1', 'P2', 1175, '4mnesia'], ['P1', 'P3', 2716, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 2290, 'DTFNODE75'], ['P1', 'P8', 100, 'TestMyTor721'], ['P1', 'P5', 100, 'artikel10buc07'], ['P2', 'P4', 885, 'LupusreginaBeta'], ['P2', 'P5', 100, 'bauruine'], ['P4', 'P5', 783, 'Cornball'], ['P4', 'P7', 100, 'schmortor'], ['P4', 'P8', 100, 'tirz'], ['P5', 'P7', 1904, 'RMK'], ['P5', 'P8', 100, 'DoubleX'], ['P3', 'P5', 100, 'orangeneis'], ['P7', 'P8', 2680, 'Vulnerar'], ['P3', 'P7', 2514, 'honestmistake'], ['P2', 'P7', 100, 'ledevin']] , {'P4'}
2023-03-26 09:04:35,553 - WARNING - [Connect-DoubleX] - Got SENDME
2023-03-26 09:04:35,556 - WARNING - [Connect-DoubleX] - Got SENDME
2023-03-26 09:04:38,340 - ERROR - [Connect-DoubleX] - Got RELAY_END
2023-03-26 09:04:41,407 - ERROR - [Connect-TestMyTor721] - Got RELAY_END
2023-03-26 09:04:44,788 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:45,370 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:45,371 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:45,384 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:46,427 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:46,531 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:46,576 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:46,604 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:46,703 - INFO - [Establish-IceMateria] - Got REND2 message from IceMateria
2023-03-26 09:04:46,703 - INFO - [Establish-IceMateria] - Entered circuit context with id 0x80000009
2023-03-26 09:04:46,703 - INFO - [Establish-IceMateria] - Stream #8: creating attached to #80000009 circuit...
2023-03-26 09:04:46,703 - INFO - [Establish-IceMateria] - Stream #8: connecting to ('127.0.0.1', 5000)
2023-03-26 09:04:47,020 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:47,260 - INFO - [RecvLoop_97.113.] - Stream #8: connected (remote ip '127.0.0.1')
2023-03-26 09:04:47,982 - INFO - [Establish-IceMateria] - Sending HELLO to peer
2023-03-26 09:04:47,982 - INFO - [Establish-IceMateria] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P8\x94ua.'
2023-03-26 09:04:50,444 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:04:50,499 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:04:50,572 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:52,559 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:54,715 - INFO - [RECEIVER] - updateLSADB: removing {'P2', 'P6', 'P4', 'P3'}
2023-03-26 09:04:54,715 - INFO - [RECEIVER] - updateLSADB: removing {'P2', 'P6', 'P4', 'P3'}
2023-03-26 09:04:54,715 - INFO - [RECEIVER] - updateLSADB: removing {'P2', 'P6', 'P4', 'P3'}
2023-03-26 09:04:54,715 - INFO - [RECEIVER] - updateGraphOnly: [['P1', 'P2', 1175, '4mnesia'], ['P1', 'P3', 2716, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 2290, 'DTFNODE75'], ['P1', 'P8', 100, 'TestMyTor721'], ['P1', 'P5', 100, 'artikel10buc07'], ['P2', 'P4', 885, 'LupusreginaBeta'], ['P2', 'P5', 1861, 'bauruine'], ['P2', 'P7', 100, 'ledevin'], ['P4', 'P5', 1920, 'Cornball'], ['P5', 'P7', 1904, 'RMK'], ['P5', 'P8', 100, 'DoubleX'], ['P3', 'P5', 100, 'orangeneis'], ['P7', 'P8', 2680, 'Vulnerar'], ['P3', 'P7', 2514, 'honestmistake'], ['P4', 'P7', 100, 'schmortor'], ['P3', 'P8', 745, 'gGDHjdcC6zAlM8k08lX'], ['P4', 'P8', 100, 'tirz']] , {'P2', 'P6', 'P4', 'P3'}
2023-03-26 09:04:55,605 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:57,360 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:57,380 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:57,380 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:57,710 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:04:57,922 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:00,491 - INFO - [RECEIVER] - updateGraphOnly: [['P1', 'P2', 1175, '4mnesia'], ['P1', 'P3', 2716, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 2290, 'DTFNODE75'], ['P1', 'P8', 100, 'TestMyTor721'], ['P1', 'P5', 100, 'artikel10buc07'], ['P4', 'P5', 1920, 'Cornball'], ['P5', 'P7', 1904, 'RMK'], ['P2', 'P5', 100, 'bauruine'], ['P5', 'P8', 100, 'DoubleX'], ['P3', 'P5', 100, 'orangeneis'], ['P7', 'P8', 2680, 'Vulnerar'], ['P4', 'P7', 100, 'schmortor'], ['P2', 'P7', 100, 'pIetr0d1C4rB04711']] , {'P2', 'P3'}
2023-03-26 09:05:01,392 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:01,432 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:02,363 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:03,378 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:05,694 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:05,736 - INFO - [RECEIVER] - updateGraphOnly: [['P1', 'P2', 1175, '4mnesia'], ['P1', 'P3', 2716, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 2290, 'DTFNODE75'], ['P1', 'P8', 100, 'TestMyTor721'], ['P1', 'P5', 100, 'artikel10buc07'], ['P4', 'P5', 1920, 'Cornball'], ['P5', 'P7', 1904, 'RMK'], ['P2', 'P5', 100, 'bauruine'], ['P5', 'P8', 100, 'DoubleX'], ['P3', 'P5', 100, 'orangeneis'], ['P7', 'P8', 2680, 'Vulnerar'], ['P4', 'P7', 100, 'schmortor'], ['P2', 'P7', 100, 'pIetr0d1C4rB04711'], ['P4', 'P8', 100, 'tirz']] , {'P8'}
2023-03-26 09:05:06,760 - INFO - [RecvLoop_192.42.] - On destroy: circuit #8000000a
2023-03-26 09:05:06,760 - INFO - [RecvLoop_192.42.] - Destroy circuit #8000000a
2023-03-26 09:05:06,875 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:10,343 - INFO - [RECEIVER] - Added neighbour P2
2023-03-26 09:05:10,343 - INFO - [RECEIVER] - updateLSADB: removing {'P7', 'P4'}
2023-03-26 09:05:10,343 - INFO - [RECEIVER] - updateLSADB: removing {'P7', 'P4'}
2023-03-26 09:05:10,343 - INFO - [RECEIVER] - updateGraphOnly: [['P7', 'P8', 1772, 'Vulnerar'], ['P3', 'P8', 1640, 'gGDHjdcC6zAlM8k08lX'], ['P1', 'P8', 100, 'TestMyTor721'], ['P5', 'P8', 100, 'DoubleX'], ['P4', 'P8', 100, 'tirz'], ['P2', 'P8', 100, 'IceMateria']] , {'P7', 'P4'}
2023-03-26 09:05:11,350 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:11,441 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:11,442 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:12,046 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:13,701 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:14,845 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:15,348 - INFO - [CONNECTION] - Neighbour connections (6) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:05:15,348 - INFO - [CONNECTION] - Candidate peers: []
2023-03-26 09:05:15,348 - INFO - [CONNECTION] - Highest cost peers: [('P7', 2004), ('P6', 1424), ('P4', 300), ('P2', 200), ('P3', 200), ('P1', 200), ('P5', 100), ('P8', 0)]
2023-03-26 09:05:15,348 - INFO - [CONNECTION] - No join peers found
2023-03-26 09:05:15,359 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:16,355 - INFO - [Thread-4] - Adding P1 to list of inactive
2023-03-26 09:05:16,358 - INFO - [Thread-4] - Adding P5 to list of inactive
2023-03-26 09:05:16,368 - INFO - [Thread-4] - DEAD ROUTES DETECTED: P1,P6,P5
2023-03-26 09:05:16,368 - INFO - [Thread-4] - Removing P1 from neighbours
2023-03-26 09:05:16,368 - INFO - [Thread-4] - updateLSADB: removing {'P1', 'P6', 'P5'}
2023-03-26 09:05:16,368 - INFO - [Thread-4] - updateLSADB: removing {'P1', 'P6', 'P5'}
2023-03-26 09:05:16,368 - INFO - [Thread-4] - updateGraphOnly: [['P1', 'P2', 1175, '4mnesia'], ['P1', 'P3', 2716, 'RMK'], ['P1', 'P4', 100, 'jdvnsjfnHBDkScf'], ['P1', 'P7', 2290, 'DTFNODE75'], ['P1', 'P5', 100, 'artikel10buc07'], ['P4', 'P5', 1920, 'Cornball'], ['P5', 'P7', 1904, 'RMK'], ['P2', 'P5', 100, 'bauruine'], ['P5', 'P8', 100, 'DoubleX'], ['P3', 'P5', 100, 'orangeneis'], ['P3', 'P8', 745, 'gGDHjdcC6zAlM8k08lX']] , {'P1', 'P6', 'P5'}
2023-03-26 09:05:16,416 - ERROR - [TESTER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:16,418 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:16,488 - ERROR - [Thread-4] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:16,622 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:16,636 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:17,120 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:19,350 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:19,637 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:20,813 - ERROR - [SENDER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:22,356 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:22,365 - INFO - [RECEIVER] - updateGraphOnly: [['P3', 'P8', 745, 'gGDHjdcC6zAlM8k08lX'], ['P2', 'P4', 885, 'LupusreginaBeta'], ['P2', 'P7', 100, 'pIetr0d1C4rB04711']] , {'P7', 'P8'}
2023-03-26 09:05:22,786 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:22,803 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:24,377 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:24,771 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:24,773 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:25,393 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:25,395 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:25,395 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:25,399 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:27,364 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:27,532 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:27,711 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:28,399 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:31,471 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,483 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,492 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,592 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,598 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,537 - INFO - [RECEIVER] - updateLSADB: removing {'P2', 'P7', 'P3'}
2023-03-26 09:05:31,600 - INFO - [RECEIVER] - updateLSADB: removing {'P2', 'P7', 'P3'}
2023-03-26 09:05:31,600 - INFO - [RECEIVER] - updateGraphOnly: [['P3', 'P8', 745, 'gGDHjdcC6zAlM8k08lX'], ['P2', 'P4', 885, 'LupusreginaBeta'], ['P2', 'P6', 2262, 'AlexandruIoanCuza2'], ['P2', 'P8', 100, 'IceMateria']] , {'P2', 'P7', 'P3'}
2023-03-26 09:05:31,603 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,605 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,605 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:31,676 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:32,247 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:32,456 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:33,668 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:33,707 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:33,961 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:35,384 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:35,403 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:35,733 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:35,876 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:35,879 - ERROR - [SENDER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:36,384 - WARNING - [RecvLoop_97.113.] - CellPadding has some unnecessary data: b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
2023-03-26 09:05:36,384 - ERROR - [RecvLoop_97.113.] - CellPadding() was received but no handlers for it
2023-03-26 09:05:37,384 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:37,421 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:37,422 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:37,474 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:37,555 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:37,534 - INFO - [RECEIVER] - updateLSADB: removing {'P6', 'P4', 'P8', 'P2', 'P3'}
2023-03-26 09:05:37,588 - INFO - [RECEIVER] - updateGraphOnly: [['P1', 'P4', 532, 'jdvnsjfnHBDkScf'], ['P4', 'P8', 100, 'tirz']] , {'P6', 'P4', 'P8', 'P2', 'P3'}
2023-03-26 09:05:38,076 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:38,218 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:40,636 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:41,420 - ERROR - [Thread-155] - networkx.exception.NodeNotFound: Node P8 not found in graph
2023-03-26 09:05:42,371 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:42,392 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:42,394 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:42,627 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:42,686 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:42,848 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:42,852 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:43,060 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:45,405 - ERROR - [Establish-Vulnerar] - recv timeout
2023-03-26 09:05:45,405 - INFO - [Establish-Vulnerar] - Closing connection Establish-Vulnerar
2023-03-26 09:05:45,405 - INFO - [Establish-Vulnerar] - Stream #2: closing (state = Connected)...
2023-03-26 09:05:45,455 - ERROR - [Thread-157] - networkx.exception.NodeNotFound: Node P8 not found in graph
2023-03-26 09:05:45,491 - INFO - [Establish-Vulnerar] - Destroy circuit #80000001
2023-03-26 09:05:45,491 - INFO - [Establish-Vulnerar] - Closing guard connections (None)...
2023-03-26 09:05:47,644 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:05:47,693 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:47,692 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:48,788 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:49,019 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:49,261 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:50,903 - WARNING - [SENDER] - socket already closed
2023-03-26 09:05:50,903 - ERROR - [SENDER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:53,368 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:05:53,371 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:53,595 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:53,596 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:05:53,806 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:55,358 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P5'
2023-03-26 09:05:58,482 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:58,504 - INFO - [Thread-4] - Adding P4 to list of inactive
2023-03-26 09:05:58,505 - INFO - [Thread-4] - DEAD ROUTES DETECTED: P4,P1,P6,P5
2023-03-26 09:05:58,505 - INFO - [Thread-4] - Removing P5 from neighbours
2023-03-26 09:05:58,505 - INFO - [Thread-4] - updateGraphOnly: [['P3', 'P8', 1895, 'gGDHjdcC6zAlM8k08lX']] , {'P4', 'P1', 'P6', 'P5'}
2023-03-26 09:05:58,522 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:05:58,525 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:05:58,623 - WARNING - [Thread-4] - socket already closed
2023-03-26 09:05:58,626 - ERROR - [Thread-4] - send_to_stream: Key not found: 'P4'
2023-03-26 09:05:58,696 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:58,696 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:05:58,982 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:00,610 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:01,195 - INFO - [RecvLoop_97.113.] - On destroy: circuit #80000009
2023-03-26 09:06:01,195 - INFO - [RecvLoop_97.113.] - Destroy circuit #80000009
2023-03-26 09:06:01,196 - INFO - [RecvLoop_97.113.] - Stream #8: closing (state = Connected)...
2023-03-26 09:06:02,390 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:03,761 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:06:03,924 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:03,932 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:04,040 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:06,362 - WARNING - [SENDER] - socket already closed
2023-03-26 09:06:06,423 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:06,651 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:07,364 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:07,368 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:09,402 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:09,402 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:10,513 - ERROR - [RecvLoop_97.113.] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-26 09:06:11,447 - ERROR - [Establish-tirz] - recv timeout
2023-03-26 09:06:11,447 - INFO - [Establish-tirz] - Closing connection Establish-tirz
2023-03-26 09:06:11,447 - INFO - [Establish-tirz] - Stream #6: closing (state = Connected)...
2023-03-26 09:06:11,472 - INFO - [Establish-tirz] - Destroy circuit #80000007
2023-03-26 09:06:11,472 - INFO - [Establish-tirz] - Closing guard connections (None)...
2023-03-26 09:06:14,414 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:14,435 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:15,423 - INFO - [CONNECTION] - Neighbour connections (4) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:06:15,423 - INFO - [CONNECTION] - Candidate peers: []
2023-03-26 09:06:15,423 - INFO - [CONNECTION] - Highest cost peers: [('P6', 2362), ('P7', 2004), ('P3', 1895), ('P1', 632), ('P5', 100), ('P4', 100), ('P2', 100), ('P8', 0)]
2023-03-26 09:06:15,423 - INFO - [CONNECTION] - No join peers found
2023-03-26 09:06:15,527 - INFO - [RECEIVER] - updateGraphOnly: [['P3', 'P8', 1895, 'gGDHjdcC6zAlM8k08lX']] , {'P7', 'P8'}
2023-03-26 09:06:15,596 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:06:19,608 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:20,455 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:20,468 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:21,464 - WARNING - [SENDER] - socket already closed
2023-03-26 09:06:21,653 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:25,668 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:25,668 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:29,416 - INFO - [RECEIVER] - updateGraphOnly: [['P3', 'P8', 1895, 'gGDHjdcC6zAlM8k08lX'], ['P2', 'P5', 1861, 'bauruine']] , {'P7', 'P4', 'P8'}
2023-03-26 09:06:29,461 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:06:30,678 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:30,678 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:32,223 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:35,696 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:36,372 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:36,681 - WARNING - [SENDER] - socket already closed
2023-03-26 09:06:36,682 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:39,364 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:41,381 - ERROR - [Establish-IceMateria] - recv timeout
2023-03-26 09:06:41,381 - INFO - [Establish-IceMateria] - Closing connection Establish-IceMateria
2023-03-26 09:06:41,381 - INFO - [Establish-IceMateria] - Stream #8: closing (state = Closed)...
2023-03-26 09:06:41,382 - WARNING - [Establish-IceMateria] - Stream #8: closed already
2023-03-26 09:06:41,382 - INFO - [Establish-IceMateria] - Destroy circuit #80000009
2023-03-26 09:06:41,382 - WARNING - [Establish-IceMateria] - #80000009 circuit: has been destroyed already
2023-03-26 09:06:41,382 - INFO - [Establish-IceMateria] - Closing guard connections (None)...
2023-03-26 09:06:41,388 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:41,389 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:46,420 - WARNING - [RecvLoop_97.113.] - CellPadding has some unnecessary data: b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
2023-03-26 09:06:46,420 - ERROR - [RecvLoop_97.113.] - CellPadding() was received but no handlers for it
2023-03-26 09:06:46,420 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:46,424 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:46,454 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:46,490 - INFO - [RECEIVER] - updateLSADB: removing {'P6', 'P4', 'P8', 'P2', 'P1', 'P3'}
2023-03-26 09:06:46,490 - INFO - [RECEIVER] - updateLSADB: removing {'P6', 'P4', 'P8', 'P2', 'P1', 'P3'}
2023-03-26 09:06:46,490 - INFO - [RECEIVER] - updateGraphOnly: [['P3', 'P8', 1895, 'gGDHjdcC6zAlM8k08lX']] , {'P6', 'P4', 'P8', 'P2', 'P1', 'P3'}
2023-03-26 09:06:47,519 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:49,480 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:06:51,460 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:51,462 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:51,520 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:51,564 - ERROR - [Thread-169] - networkx.exception.NodeNotFound: Node P8 not found in graph
2023-03-26 09:06:51,758 - WARNING - [SENDER] - socket already closed
2023-03-26 09:06:51,798 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:51,799 - WARNING - [SENDER] - socket already closed
2023-03-26 09:06:58,375 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:58,375 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:06:58,375 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:06:59,635 - INFO - [RECEIVER] - updateLSADB: removing {'P1', 'P7', 'P4'}
2023-03-26 09:06:59,656 - INFO - [RECEIVER] - updateGraphOnly: [] , {'P1', 'P7', 'P4'}
2023-03-26 09:06:59,868 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:07:00,878 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:02,382 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:07:03,490 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:03,493 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:03,494 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:07,387 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:07,416 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:07,416 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:09,460 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:09,460 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:09,460 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:14,817 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:14,819 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:14,822 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:16,090 - INFO - [CONNECTION] - Neighbour connections (4) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:07:16,092 - INFO - [CONNECTION] - Candidate peers: []
2023-03-26 09:07:16,092 - INFO - [CONNECTION] - Highest cost peers: [('P6', 2362), ('P7', 2004), ('P3', 1895), ('P1', 632), ('P5', 100), ('P4', 100), ('P2', 100), ('P8', 0)]
2023-03-26 09:07:16,096 - INFO - [CONNECTION] - No join peers found
2023-03-26 09:07:20,380 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:20,380 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:20,380 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:22,461 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:22,462 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:22,464 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:24,704 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:07:25,461 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:25,472 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:25,472 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:25,713 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:26,754 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:07:28,672 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:07:28,831 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:07:31,529 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:31,531 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:31,532 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:36,545 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:36,547 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:36,547 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:38,400 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:38,407 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:38,492 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:41,589 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:41,592 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:41,592 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:46,935 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:46,936 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:46,936 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:52,437 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:52,556 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:52,559 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:54,674 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:54,676 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:54,676 - WARNING - [SENDER] - socket already closed
2023-03-26 09:07:57,629 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:07:57,709 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:07:57,745 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:03,477 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:03,483 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:03,483 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:08,509 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:08,650 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:08,728 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:09,968 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:10,087 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:10,113 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:14,411 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:14,414 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:14,415 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:16,430 - INFO - [CONNECTION] - Neighbour connections (4) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:08:16,430 - INFO - [CONNECTION] - Candidate peers: []
2023-03-26 09:08:16,430 - INFO - [CONNECTION] - Highest cost peers: [('P6', 2362), ('P7', 2004), ('P3', 1729), ('P1', 632), ('P5', 100), ('P4', 100), ('P2', 100), ('P8', 0)]
2023-03-26 09:08:16,430 - INFO - [CONNECTION] - No join peers found
2023-03-26 09:08:19,440 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:19,441 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:19,444 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:24,453 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:24,458 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:24,460 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:25,428 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:25,477 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:25,477 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:27,083 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:08:28,401 - ERROR - [RECEIVER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:29,416 - WARNING - [RECEIVER] - socket already closed
2023-03-26 09:08:29,500 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:30,409 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:30,412 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:36,627 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:36,815 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:37,498 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:40,533 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:40,730 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:40,731 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:42,557 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:42,558 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:42,558 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:47,576 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:47,713 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:47,727 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:53,153 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:53,245 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:53,248 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:56,286 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:56,287 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:56,288 - WARNING - [SENDER] - socket already closed
2023-03-26 09:08:58,410 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:58,533 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:08:58,534 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:08:58,816 - WARNING - [Connect-gGDHjdcC6zAlM8k08lX] - Got SENDME
2023-03-26 09:09:03,760 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:09:03,763 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:09:03,763 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:09:08,781 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:09:08,844 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:09:08,845 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:09:11,536 - WARNING - [SENDER] - socket already closed
2023-03-26 09:09:11,536 - ERROR - [SENDER] - send_to_stream: Key not found: 'P4'
2023-03-26 09:09:11,537 - WARNING - [SENDER] - socket already closed
2023-03-26 09:09:14,032 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:09:14,033 - ERROR - [HEART BEAT] - send_to_stream: Key not found: 'P4'
2023-03-26 09:09:14,033 - WARNING - [HEART BEAT] - socket already closed
2023-03-26 09:09:16,526 - INFO - [CONNECTION] - Neighbour connections (4) less than MIN_NEIGHBOUR_CONNECTIONS (8)
2023-03-26 09:09:16,527 - INFO - [CONNECTION] - Candidate peers: []
2023-03-26 09:09:16,527 - INFO - [CONNECTION] - Highest cost peers: [('P6', 2362), ('P7', 2004), ('P1', 632), ('P3', 522), ('P5', 100), ('P4', 100), ('P2', 100), ('P8', 0)]
2023-03-26 09:09:16,527 - INFO - [CONNECTION] - No join peers found
